cmd/soroban-rpc/internal/db/event.go

```diff
@@ -97,3 +112,225 @@ func (eventHandler *eventHandler) InsertEvents(lcm xdr.LedgerCloseMeta) error {
 
 	return nil
 }
+
+type ScanFunction func(xdr.DiagnosticEvent, events.Cursor, int64, *xdr.Hash) bool
```

Comment:
Please add parameter names (or use more specific types) so that the reader can understand what are the `int64` and `*xdr.Hash` parameters.